### ISSUE TITLE
neigborhood_size

### ISSUE THREAD

**Issue by Fly-Yun-Maker** [s4mcontext/s4mc#1]
Hello, thank you for your excellent work, I'm a little confused as to why the size of the neighbors in the code is 4×4 and why the size in the paper is 3×3

**Comment by s4mcontext** [comment]
Hi,

Please note that in the provided code the neighborhood size is always 3x3!

We use a padding trick:
`X = torch.nn.functional.pad(X, (1, 1, 1, 1, 0, 0, 0, 0))`

If you refer to the arg 'n' for `def get_neigbor_tensors()`, it refer to the number of neighbors, that is fixed to either 4 or 8.

Upon acceptance we will release the code for the experiments that achieve SOTA results, as well as more models, datasets and varying neighborhood size

**Comment by s4mcontext** [comment]
Another comment, the padding trick make it very fast to calculate the joint probability for so many pixels with a relevantly low memory cost, we really don't recommend to implement it with iterating over pixels.
